syntax = "proto3";

message ProtoView {
    int32 xleft = 1;
    int32 ytop = 2;
    int32 xright = 3;
    int32 ybottom = 4;

    string type = 5;
    string id = 6;

    int32 seq_id = 8;
    int32 parent_seq_id = 9;
    repeated int32 children_ids = 10;

    bool is_content_node = 11;

    map<string, string> properties = 12;
}

message ProtoScreen {
    string window_path = 1;

    string theme = 2;
    string device = 3;
    int32 device_width = 4;
    int32 device_height = 5;

    repeated ProtoView views = 6;
}

message ProtoApp {
    string package_name = 1;
    string file_name = 2;
    repeated ProtoScreen screens = 3;
}

//message ProtoApp {
//    string name = 1;
//    string window_path = 2;
//
//    repeated ProtoView views = 3;
//
//    int32 device_width = 4;
//    int32 device_height = 5;
//}